runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/DataflowWorkerHarnessHelper.java

```diff
@@ -95,6 +97,17 @@ public static void configureLogging(DataflowWorkerHarnessOptions pipelineOptions
     DataflowWorkerLoggingInitializer.configure(pipelineOptions);
   }
 
+  public static void runUserDefinedInitialization() {
+    ServiceLoader<DataflowWorkerInitializer> loader =
+        ServiceLoader.load(DataflowWorkerInitializer.class);
+    for (DataflowWorkerInitializer initializer : loader) {
```

Comment:
You should be able to write a test with a Test "initializer" that gets detected, loaded and performs some detectable side effect. I think there are several examples for serviceloader tests in the code base already.

Reply:
I moved all the relevant logic into BeamWorkerInitializer and tested there. It's kind of tricky to test the worker harnesses directly.